drivers/entropy/entropy_bt_hci.c

```diff
@@ -20,9 +20,18 @@ static int entropy_bt_init(const struct device *dev)
 static int entropy_bt_get_entropy(const struct device *dev,
 				  uint8_t *buffer, uint16_t length)
 {
-	if (!bt_is_ready()) {
-		return -EAGAIN;
-	}
+	/* Do not wait for BT to be ready (i.e. bt_is_ready()) before issueing
+	 * the command. The reason is that when crypto is enabled and the PSA
+	 * Crypto API support is provided through Mbed TLS, random number generator
+	 * needs to be available since the very first call to psa_crypto_init()
+	 * which is usually done before BT is completely initialized.
+	 * On the other hand, in devices like the nrf5340, the crytographically
+	 * secure RNG is owned by the cpu_net, so the cpu_app needs to poll it
+	 * to get random data. Again, there is no need to wait for BT to be
+	 * completely initialized for this kind of support. Just try to send the
+	 * request through HCI. If the command fails for any reason, then
+	 * we return failure anyway.
+	 */
```

Comment:
As the return value could be different, we need API documentation change?

Reply:
I added the documentation to the `migration-guide-4.0` document. I know it's not the right place because this won't be part of 4.0, but in the current status of the PR there's no 4.1 version of that document yet and rebasing the PR is a bit too much IMO just for this change. I will move the documentation to the proper document once
I can finalize the PR (i.e. solving all the issues with the CI).